src/main/java/io/percy/appium/AppPercy.java

```diff
@@ -107,12 +109,16 @@ public JSONObject screenshot(String name, Boolean fullScreen, ScreenshotOptions
         }
         percyOptions.setPercyIgnoreErrors();
         try {
-            GenericProvider provider = ProviderResolver.resolveProvider(driver);
             if (options == null) {
                 options = new ScreenshotOptions();
             }
             options.setFullScreen(fullScreen);
-            return provider.screenshot(name, options).getJSONObject("data");
+            JSONObject response = provider.screenshot(name, options);
+            if (response != null && response.has("data")) {
```

Comment:
in [src/main/java/io/percy/appium/providers/AppAutomate.java](https://github.com/percy/percy-appium-java/blob/a78cdbf94a5deb1c729835a8d5f0ea65363529bc/src/main/java/io/percy/appium/providers/AppAutomate.java#L172) Remove the return of data. If we are doing it here we can just return response from there. No need to have to return data in 2 places.

Reply:
got it. 😵‍💫